scalardb-test/src/main/java/kelpie/scalardb/transfer/graphql/ThreadLocalCookieJar.java

```diff
@@ -0,0 +1,43 @@
+package kelpie.scalardb.transfer.graphql;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import okhttp3.Cookie;
+import okhttp3.CookieJar;
+import okhttp3.HttpUrl;
+import org.jetbrains.annotations.NotNull;
+
+/** A {@code CookieJar} implementation that manages cookies for each thread. */
+public class ThreadLocalCookieJar implements CookieJar {
+
+  private static final ThreadLocal<List<Cookie>> cookies = ThreadLocal.withInitial(ArrayList::new);
+
+  @NotNull
+  @Override
+  public List<Cookie> loadForRequest(@NotNull HttpUrl httpUrl) {
+    List<Cookie> validCookies = new ArrayList<>();
+    for (Iterator<Cookie> it = cookies.get().iterator(); it.hasNext(); ) {
+      Cookie currentCookie = it.next();
+      if (isCookieExpired(currentCookie)) {
+        it.remove();
+      } else if (currentCookie.matches(httpUrl)) {
+        validCookies.add(currentCookie);
+      }
+    }
+    return validCookies;
+  }
+
+  private static boolean isCookieExpired(Cookie cookie) {
+    return cookie.expiresAt() < System.currentTimeMillis();
+  }
+
+  @Override
+  public void saveFromResponse(@NotNull HttpUrl httpUrl, @NotNull List<Cookie> list) {
+    cookies.get().addAll(list);
```

Comment:
If a HTTP client is reused and receives a response multiple times, the number of elements in this list increases. Maybe we need to update Cookie in `cookies` if it already exists?

Reply:
Okay. I thought we didn't need to clear the same cookie here because in our use case we clear all the cookies per transaction. But, okay, let me do that. Thanks.

Reply:
Fixed in 0f4264be51f5282542ee0f2a1ff809d4a66711b3. Thanks.